MLIR.cmake

```diff
@@ -70,11 +70,20 @@ function(add_onnx_mlir_dialect_doc dialect dialect_tablegen_file)
 endfunction()
 add_custom_target(onnx-mlir-docs)
 
-function(add_onnx_mlir_dialect dialect)
-  set(LLVM_TARGET_DEFINITIONS ${dialect}.td)
-  mlir_tablegen(${dialect}.hpp.inc -gen-op-decls "-I${ONNX_MLIR_SRC_ROOT}")
-  mlir_tablegen(${dialect}.cpp.inc -gen-op-defs "-I${ONNX_MLIR_SRC_ROOT}")
-  add_public_tablegen_target(OM${dialect}IncGen)
+# If an extra parameter, the dialect name, is provided,
+# this function will generate dialect and type from the td file
+function(add_onnx_mlir_dialect tdfile)
+  set(LLVM_TARGET_DEFINITIONS ${tdfile}.td)
+  mlir_tablegen(${tdfile}Ops.hpp.inc -gen-op-decls "-I${ONNX_MLIR_SRC_ROOT}")
+  mlir_tablegen(${tdfile}Ops.cpp.inc -gen-op-defs "-I${ONNX_MLIR_SRC_ROOT}")
+  if (${ARGC} EQUAL 2) 
```

Comment:
Rather than just checking by number of arguments, you can have "proper" named arguments and let cmake parse them for you. It's a lot less prone to error.